lib/Backend/Func.cpp

```diff
@@ -978,7 +978,7 @@ bool Func::CanAllocInPreReservedHeapPageSegment ()
 {
 #ifdef _CONTROL_FLOW_GUARD
     return PHASE_FORCE1(Js::PreReservedHeapAllocPhase) || (!PHASE_OFF1(Js::PreReservedHeapAllocPhase) &&
-        !IsJitInDebugMode() && !GetScriptContext()->IsInDebugMode() && GetScriptContext()->GetThreadContext()->IsCFGEnabled()
+        !IsJitInDebugMode() && !this->m_workItem->GetFunctionBody()->GetUtf8SourceInfo()->IsUtf8SourceInfoInDebugMode() && GetScriptContext()->GetThreadContext()->IsCFGEnabled()
```

Comment:
When a CodeGenWorkItem is created we set if it is created in debug mode or not, See NativeCodeGenerator::NewFunctionCodeGen or NativeCodeGenerator::NewLoopBodyCodeGen.